x-pack/plugins/spaces/server/capabilities/capabilities_switcher.ts

```diff
@@ -4,15 +4,37 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import _ from 'lodash';
-import { UICapabilities } from 'ui/capabilities';
+import { Capabilities, CapabilitiesSwitcher, CoreSetup } from 'src/core/server';
 import { Feature } from '../../../../plugins/features/server';
 import { Space } from '../../common/model/space';
+import { SpacesServiceSetup } from '../spaces_service';
+import { PluginsSetup } from '../plugin';
 
-export function toggleUICapabilities(
-  features: Feature[],
-  capabilities: UICapabilities,
-  activeSpace: Space
-) {
+export function setupCapabilitiesSwitcher(
+  core: CoreSetup<PluginsSetup>,
+  spacesService: SpacesServiceSetup
+): CapabilitiesSwitcher {
+  return async (request, capabilities) => {
+    const isAnonymousRequest = !request.route.options.authRequired;
+
+    if (isAnonymousRequest) {
+      return capabilities;
```

Comment:
The existing implementation behaved this way, but it wasn't explicit. We used to silently return the capabilities unmodified if an error was thrown here. If an unauthenticated route attempts to retrieve the active space, the request fails because auth headers aren't attached to the request.

Reply:
Yeah, I agree, thanks for explaining!